client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -167,72 +173,183 @@ public Response performRequest(String method, String endpoint, Map<String, Strin
+        return listener.get();
+    }
+
+    /**
+     * Sends a request to the elasticsearch cluster that the client points to. Doesn't wait for the response, instead
+     * the provided {@link ResponseListener} will be notified upon completion or failure. Shortcut to
+     * {@link #performRequest(String, String, Map, HttpEntity, ResponseListener, Header...)} but without parameters and  request body.
+     *
+     * @param method the http method
+     * @param endpoint the path of the request (without host and port)
+     * @param responseListener the {@link ResponseListener} to notify when the request is completed or fails
+     * @param headers the optional request headers
+     */
+    public void performRequest(String method, String endpoint, ResponseListener responseListener, Header... headers) {
+        performRequest(method, endpoint, Collections.<String, String>emptyMap(), null, responseListener, headers);
+    }
+
+    /**
+     * Sends a request to the elasticsearch cluster that the client points to. Doesn't wait for the response, instead
+     * the provided {@link ResponseListener} will be notified upon completion or failure. Shortcut to
+     * {@link #performRequest(String, String, Map, HttpEntity, ResponseListener, Header...)} but without request body.
+     *
+     * @param method the http method
+     * @param endpoint the path of the request (without host and port)
+     * @param params the query_string parameters
+     * @param responseListener the {@link ResponseListener} to notify when the request is completed or fails
+     * @param headers the optional request headers
+     */
+    public void performRequest(String method, String endpoint, Map<String, String> params,
+                               ResponseListener responseListener, Header... headers) {
+        performRequest(method, endpoint, params, null, responseListener, headers);
+    }
+
+    /**
+     * Sends a request to the elasticsearch cluster that the client points to. Doesn't wait for the response, instead
+     * the provided {@link ResponseListener} will be notified upon completion or failure.
+     * Shortcut to {@link #performRequest(String, String, Map, HttpEntity, HttpAsyncResponseConsumer, ResponseListener, Header...)}
+     * which doesn't require specifying an {@link HttpAsyncResponseConsumer} instance, {@link HeapBufferedAsyncResponseConsumer}
+     * will be used to consume the response body.
+     *
+     * @param method the http method
+     * @param endpoint the path of the request (without host and port)
+     * @param params the query_string parameters
+     * @param entity the body of the request, null if not applicable
+     * @param responseListener the {@link ResponseListener} to notify when the request is completed or fails
+     * @param headers the optional request headers
+     */
+    public void performRequest(String method, String endpoint, Map<String, String> params,
```

Comment:
I tend to agree in principle, but more `performRequest` methods were added following a discussion on FixItFriday, see #19312, to not require people to pass in null arguments and empty maps all the time.